iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/integration/test/IcebergRestKerberosHiveCatalogIT.java

```diff
@@ -149,12 +152,50 @@ Map<String, String> getCatalogConfig() {
     return configMap;
   }
 
-  private static boolean isEmbedded() {
-    String mode =
-        System.getProperty(ITUtils.TEST_MODE) == null
-            ? ITUtils.EMBEDDED_TEST_MODE
-            : System.getProperty(ITUtils.TEST_MODE);
+  protected void initSparkEnv() {
+    int port = getServerPort();
+    LOG.info("Iceberg REST server port:{}", port);
+    String icebergRESTUri = String.format("http://127.0.0.1:%d/iceberg/", port);
+    SparkConf sparkConf =
+        new SparkConf()
+            .set(
+                "spark.sql.extensions",
+                "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions")
+            .set("spark.sql.catalog.rest", "org.apache.iceberg.spark.SparkCatalog")
+            .set("spark.sql.catalog.rest.type", "rest")
+            .set("spark.sql.catalog.rest.uri", icebergRESTUri)
+            .set("spark.locality.wait.node", "0");
+
+    if (supportsCredentialVending()) {
+      sparkConf.set(
+          "spark.sql.catalog.rest.header.X-Iceberg-Access-Delegation", "vended-credentials");
+    }
+
+    // Login kerberos and use the users to execute the spark job.
+    try {
+      Configuration configuration = new Configuration();
+      configuration.set("hadoop.security.authentication", "kerberos");
+      KerberosClient kerberosClient =
+          new KerberosClient(
+              ImmutableMap.of(
+                  "authentication.kerberos.principal",
+                  HIVE_METASTORE_CLIENT_PRINCIPAL,
+                  "authentication.kerberos.keytab-uri",
+                  tempDir + HIVE_METASTORE_CLIENT_KEYTAB),
+              configuration);
+
+      kerberosClient.login(tempDir + HIVE_METASTORE_CLIENT_KEYTAB);
```

Comment:
I'm afraid the normal user couldn't do `kerberosClient.login` in `spark-sql` or `spark-shell`, may be you could try add  `spark.kerberos.principal` and `spark.kerberos.keytab` to spark config.

Reply:
OK, let me optimized it.